File: sorting/Rcode/bubbleSort.R


bubbleSort <- function(a){
  swapCount <- 0
  for(i in 1:length(a)){
    for(j in 1:(length(a)-1)){
      if(a[j] > a[(j+1)]){
        tmp1 <- a[j]
        tmp2 <- a[(j+1)]
        a[j] <- tmp2
        a[(j+1)] <- tmp1
        swapCount <- swapCount + 1
      }
    }
  }
  ans <- list(paste('Array is sorted in',swapCount,"swaps."),
              paste('First Element:',a[1]),
              paste('Last Element:',a[length(a)]))
  
  return(ans)
}

a <- c(1,2,3)
bubbleSort(a)

a <- c(3,2,1)
bubbleSort(a)

File: sorting/py_code/bubble_sort.py
def bubble_sort(a):
    swap_count = 0
    for i in range(len(a)):
        for j in range(len(a)-1):
            if a[j] > a[(j+1)]:
                tmp1 = a[j]
                tmp2 = a[(j+1)]
                a[j] = tmp2
                a[(j+1)] = tmp1
                swap_count = swap_count + 1

    return 'Array is sorted in ' + str(swap_count) + ' swaps.\n' + \
        'First Element: ' + str(a[0]) + '\n' + \
        'Last Element: ' + str(a[(len(a)-1)])


a = [3,2,1]
print(bubble_sort(a))

a = [5,4,3,2,1]
print(bubble_sort(a))
